contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -0,0 +1,720 @@
+     * @param _initCap is the initial vault's cap on deposits, the manager can increase this as necessary.
+     * @param _tokenName is the name of the vault share token
+     * @param _tokenSymbol is the symbol of the vault share token
+     */
+    function initialize(
+        address _owner,
+        address _feeRecipient,
+        uint256 _initCap,
+        string calldata _tokenName,
+        string calldata _tokenSymbol
+    ) external initializer {
+        require(_owner != address(0), "!_owner");
+        require(_feeRecipient != address(0), "!_feeRecipient");
+        require(_initCap > 0, "_initCap > 0");
+        require(bytes(_tokenName).length > 0, "_tokenName != 0x");
+        require(bytes(_tokenSymbol).length > 0, "_tokenSymbol != 0x");
+
+        __ReentrancyGuard_init();
+        __ERC20_init(_tokenName, _tokenSymbol);
+        __Ownable_init();
+        transferOwnership(_owner);
+        cap = _initCap;
+
+        // hardcode the initial withdrawal fee
+        instantWithdrawalFee = 0.005 ether;
+        feeRecipient = _feeRecipient;
+    }
+
+    /**
+     * @notice Sets the new manager of the vault.
+     * @param newManager is the new manager of the vault
+     */
+    function setManager(address newManager) external onlyOwner {
+        require(newManager != address(0), "!newManager");
+        address oldManager = manager;
+        manager = newManager;
+
+        emit ManagerChanged(oldManager, newManager);
+    }
+
+    /**
+     * @notice Sets the new fee recipient
+     * @param newFeeRecipient is the address of the new fee recipient
+     */
+    function setFeeRecipient(address newFeeRecipient) external onlyOwner {
+        require(newFeeRecipient != address(0), "!newFeeRecipient");
+        feeRecipient = newFeeRecipient;
+    }
+
+    /**
+     * @notice Sets the new withdrawal fee
+     * @param newWithdrawalFee is the fee paid in tokens when withdrawing
+     */
+    function setWithdrawalFee(uint256 newWithdrawalFee) external onlyManager {
+        require(newWithdrawalFee > 0, "withdrawalFee != 0");
+
+        // cap max withdrawal fees to 30% of the withdrawal amount
+        require(newWithdrawalFee < 0.3 ether, "withdrawalFee >= 30%");
+
+        uint256 oldFee = instantWithdrawalFee;
+        emit WithdrawalFeeSet(oldFee, newWithdrawalFee);
+
+        instantWithdrawalFee = newWithdrawalFee;
+    }
+
+    /**
+     * @notice Deposits ETH into the contract and mint vault shares. Reverts if the underlying is not WETH.
+     */
+    function depositETH() external payable nonReentrant {
+        require(asset == WETH, "asset is not WETH");
+        require(msg.value > 0, "No value passed");
+
+        IWETH(WETH).deposit{value: msg.value}();
+        _deposit(msg.value);
+    }
+
+    /**
+     * @notice Deposits the `asset` into the contract and mint vault shares.
+     * @param amount is the amount of `asset` to deposit
+     */
+    function deposit(uint256 amount) external nonReentrant {
```

Comment:
I could see the utility of a `depositYieldToken` and `withdrawYieldToken` function. It will be niche but quite useful to some people.

E.g. existing Yearn depositors want to get higher yields than what they have right now with their existing ytokens

Reply:
👍